### PR TITLE
Add doc for custom lifetime of java actor

### PR DESCRIPTION
**Pull request by tomsunelite**
## Why are these changes needed?

 Custom lifetime of java Actor is already supported, but the related document is not updated

## Related issue number

[Core] #26705

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-core/actors/named-actors.rst [hunk 1]
```diff
@@ -179,7 +178,22 @@ Actor Lifetimes
...
 
-    Customizing lifetime of an actor hasn't been implemented in Java yet.
+    .. code-block:: java
+
+        System.setProperty("ray.job.namespace", "lifetime");
+        Ray.init();
+        ActorHandle<Counter> counter = Ray.actor(Counter::new).setName("some_name").setLifetime(ActorLifetime.DETACHED).remote();
+    
+    The CounterActor will be kept alive even after the driver running above script
+    exits. Therefore it is possible to run the following script in a different
```
Comment (jovany-wang) [inline doc/source/ray-core/actors/named-actors.rst:188]:
```suggestion
    exits. Therefore it is possible to run the following code in a different
```

**Comment by jovany-wang** [comment]
Also you need to sign off the latest commit. You can use this command to commit your new changes:

`git commit -m "you commit message" -s`
And then it should be signed off.

**Comment by jovany-wang** [comment]
The change looks good overall. Please approve it as you guys are the doc code owners. @simon-mo @ericl 
I'll take over the rest, thanks.

**Review by simon-mo — APPROVED** [review]
Stamp

**Review by jovany-wang — APPROVED** [review]

**Review by WangTaoTheTonic — APPROVED** [review]

**Review by jovany-wang — APPROVED** [review]

**Comment by jovany-wang** [comment]
Thanks for your contributions!

**Pull request merged**